src/services/codefixes/helpers.ts

```diff
@@ -195,7 +195,10 @@ namespace ts.codefix {
             if (method) addClassElement(method);
         }
 
-        function createName(node: PropertyName) {
+        function createName(node: PropertyName, parent?: Declaration) {
+            if (parent && isPropertySignature(parent) && isIdentifier(node) && node.escapedText === "constructor") {

```

Comment:
why is it important to check the parent? It does restrict the change to fixing just #47954, but I don't think you ever want to emit `constructor` in a fix. I think this fix should apply to most or all of the places that `createName` is called, although that means a lot more testing.

Reply:
Do we need to handle `constructor` name for getters/setters too?